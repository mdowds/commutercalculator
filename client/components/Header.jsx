"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
function Header(props) {
    var divStyle = {
        backgroundColor: "white",
        width: "100%"
    };
    var h1Style = {
        margin: "2px 0",
        fontSize: "25px",
        textAlign: "center"
    };
    return (<div style={divStyle}>
            <h1 style={h1Style}>Commuter Calculator</h1>
            {props.children}
        </div>);
}
exports.default = Header;
