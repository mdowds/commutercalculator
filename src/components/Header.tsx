import * as React from "react";
import {CSSProperties, ReactNode} from "react";

interface HeaderProps {
  children: ReactNode;
};

export default function Header(props: HeaderProps) {
  const divStyle: CSSProperties = {
    backgroundColor: "white",
    width: "100%"
  };

  const h1Style: CSSProperties = {
    margin: "2px 0",
    fontSize: "25px",
    textAlign: "center"
  };

  return (
    <div style={divStyle}>
      <h1 style={h1Style}>Commuter Calculator</h1>
      {props.children}
    </div>
  );
}
