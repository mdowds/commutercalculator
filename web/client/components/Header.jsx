import React from 'react';

export default class Header extends React.Component {

    render() {

        const divStyle = {
            padding: "5px",
            position: "fixed",
            zIndex: 1,
            backgroundColor: "white",
            width: "100%"
        };

        const title = this.props.dest !== undefined ? "Journeys to " + this.props.dest : "Commuter Calculator";

        return (
            <div style={divStyle}>
                {title}
            </div>
        );
    }

}
