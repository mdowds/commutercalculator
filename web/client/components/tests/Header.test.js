import React from 'react';
import {shallow} from 'enzyme';
import Header from '../Header.jsx';

test("Header displays the destination", () => {
    const header = shallow(<Header dest="Waterloo" />);
    expect(header.text()).toEqual("Journeys to Waterloo")
});

test("Header displays the default text", () => {
    const header = shallow(<Header />);
    expect(header.text()).toEqual("Commuter Calculator")
});
