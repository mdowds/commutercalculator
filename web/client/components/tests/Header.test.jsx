"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var renderer = require("react-test-renderer");
var Header_1 = require("../Header");
test("Header renders the title", function () {
    var header = renderer.create(<Header_1.default />);
    var tree = header.toJSON();
    expect(tree).toMatchSnapshot();
});
test("Header renders children", function () {
    var header = renderer.create(<Header_1.default>Child content</Header_1.default>);
    var tree = header.toJSON();
    expect(tree).toMatchSnapshot();
});
