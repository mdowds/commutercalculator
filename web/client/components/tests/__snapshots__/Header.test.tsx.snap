// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders children 1`] = `
<div
  style={
    Object {
      "backgroundColor": "white",
      "width": "100%",
    }
  }
>
  <h1
    style={
      Object {
        "fontSize": "25px",
        "margin": "2px 0",
        "textAlign": "center",
      }
    }
  >
    Commuter Calculator
  </h1>
  Child content
</div>
`;

exports[`Header renders the title 1`] = `
<div
  style={
    Object {
      "backgroundColor": "white",
      "width": "100%",
    }
  }
>
  <h1
    style={
      Object {
        "fontSize": "25px",
        "margin": "2px 0",
        "textAlign": "center",
      }
    }
  >
    Commuter Calculator
  </h1>
</div>
`;
